crates/cli_testing_examples/benchmarks/NQueens.roc

```diff
@@ -40,14 +41,15 @@ appendSafe = \k, soln, solns ->
     else
         appendSafe (k - 1) soln solns
 
-
 safe : I64, I64, ConsList I64 -> Bool
 safe = \queen, diagonal, xs ->
     when xs is
         Nil -> Bool.true
         Cons q t ->
-            queen != q && queen != q + diagonal && queen != q - diagonal && safe queen (diagonal + 1) t
-
```

Comment:
it would be nice if we could auto-desugar this to hit the tail-recursion analysis, rather than relying on an explicit if/else. Filed https://github.com/roc-lang/roc/issues/5593